ckanext/geocat/harvester.py

```diff
@@ -38,150 +36,141 @@ def info(self):
+            gathered_ogdch_identifiers=gathered_ogdch_identifiers,
+        )
+        packages_that_cannot_be_created = search_utils.get_double_packages(
+            existing_dataset_infos=existing_dataset_infos,
+            gathered_ogdch_identifiers=gathered_ogdch_identifiers,
+        )
+        for identifier, info in packages_that_cannot_be_created:
+            gathered_ogdch_identifiers.remove(identifier)
+            self._save_gather_error(
+                'Unable to create package: %s since a package with this identifier already exists'  # noqa
+                % (identifier),
+                harvest_job
             )
-            return False
 
-        csw_url = harvest_object.source.url.rstrip('/')
-        csw = None
-        try:
-            csw = md.CswHelper(url=csw_url)
-            xml = csw.get_by_id(harvest_object.guid)
-            harvest_object.content = xml
-            harvest_object.save()
-            log.debug('successfully processed ' + harvest_object.guid)
-            return True
-        except Exception, e:
-            response = '-'
-            if csw and hasattr(csw.catalog, 'response'):
-                response = csw.catalog.response
+        csw_map = csw_mapping.GeoMetadataMapping(
+            organization_slug=self.config['organization'],
+            geocat_perma_link=self.config['geocat_perma_link_url'],
+            geocat_perma_label=self.config['geocat_perma_link_label'],
+            legal_basis_url=self.config['legal_basis_url'],
+            valid_identifiers=all_ogdch_identifiers,
+        )
 
-            self._save_object_error(
-                (
-                    'Unable to get content for package: %s: %r: %r / %s' %
-                    (
-                        harvest_object.guid,
-                        e,
-                        response,
-                        traceback.format_exc()
-                    )
-                ),
-                harvest_object
-            )
-            return False
+        for geocat_id in gathered_geocat_identifiers:
+
+            ogdch_identifier = ogdch_map_utils.map_geocat_to_ogdch_identifier(
+                geocat_identifier=geocat_id,
+                organization_slug=self.config['organization'])
+            csw_record_as_string = csw_data.get_record_by_id(geocat_id)
+            dataset_dict = csw_map.get_metadata(csw_record_as_string, geocat_id)  # noqa
+            try:
+                harvest_obj = HarvestObject(guid=ogdch_identifier,
+                                            job=harvest_job,
+                                            content=json.dumps(dataset_dict))
+                harvest_obj.save()
+                harvest_obj_ids.append(harvest_obj.id)
+            except Exception as e:
+                self._save_gather_error(
+                    'Error when processsing dataset: %s %r / %s'
+                    % (ogdch_identifier, e, traceback.format_exc()), harvest_job)  # noqa
+                return []
```

Comment:
Returning here means that if we get an error when importing one dataset, we don't import anything from the whole source. I don't think that's the correct behaviour. You can just delete this line and it should work correctly.